src/main/org/apache/tools/ant/taskdefs/Parallel.java

```diff
@@ -377,7 +377,7 @@ public synchronized void run() {
         }
 
         // now did any of the threads throw an exception
-        exceptionMessage = new StringBuffer();
+        exceptionMessage = new StringBuilder();
```

Comment:
Same comment as above.

Reply:
Looking through the class I don't think `exceptionMessage` needs to be an instance field at all, it could be a local variable in `spinThreads` and get passed as an argument to `processExceptions` without doing any harm. To me it seems it is only ever used by a single thread.

Most probably a further refactoring could get rid of the `first*` instance fields as well and have `processExceptions` return all their values nicely encapsulated in a single object - including the accumulated messages.